backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/DeleteByQueryPerformer.java

```diff
@@ -57,8 +58,8 @@ public DeleteByQueryPerformer(Client client, ExecutorService executor, int batch
         this.typeName = typeName;
     }
 
-    public void perform(QueryBuilder queryBuilder) {
-        executor.execute(() -> doDeleteByQuery(queryBuilder));
+    public Future<?> perform(QueryBuilder queryBuilder) {
```

Comment:
Because `doDeleteByQuery` returns a `void` and not a `Void`. I find it little useful to change this and enforce a `return null;` in that method.

(Though that could easily be done if you find it useful).

Reply:
Ok, I agree it's very ugly to `return null` but then you pass the burden to the API caller that is not able to know what the return type is. I would choose the return null, even if it doesn't matter much anyway.

Reply:
Said like this, I do agree with you.